src/parent/props.js

```diff
@@ -11,7 +11,7 @@ import type { ParentHelpers } from './index';
 export function extendProps<P, X>(propsDef : PropsDefinitionType<P, X>, existingProps : PropsType<P>, inputProps : PropsInputType<P>, helpers : ParentHelpers<P>, container : HTMLElement | void) {
     const { state, close, focus, event, onError } = helpers;
 
-    const newProps = { ...existingProps };
+    const newProps = { ...existingProps, ...inputProps };
```

Comment:
This code use to `extend(props, inputProps)` where `props` has been renamed to `currentProps`. The more I compare the existing to new the less I think this is the correct fix. I'll need to dig a bit more in the morning, but the test case and root idea is hopefully a good start for finding the right fix here.